reference-implementation/lib/readable-stream.js

```diff
@@ -700,8 +700,57 @@ class ReadableStreamDefaultReader {
 
     ReadableStreamReaderGenericRelease(this);
   }
+
+  getIterator({ preventCancel = false } = {}) {
+    if (IsReadableStreamDefaultReader(this) === false) {
+      throw defaultReaderBrandCheckException('getIterator');
+    }
+    if (this._ownerReadableStream === undefined) {
+      throw readerLockException('getIterator');
+    }
+    const iterator = Object.create(ReadableStreamDefaultReaderAsyncIteratorPrototype, {
+      _reader: { value: undefined },
+      _preventCancel: { value: undefined }
+    });
+    iterator._reader = this;
+    iterator._preventCancel = Boolean(preventCancel);
+    return iterator;
+  }
 }
 
+ReadableStreamDefaultReader.prototype[Symbol.asyncIterator] = ReadableStreamDefaultReader.prototype.getIterator;
+
+const ReadableStreamDefaultReaderAsyncIteratorPrototype = Object.setPrototypeOf({
+  next() {
+    if (!IsReadableStreamDefaultReaderAsyncIterator(this)) {
+      throw defaultReaderAsyncIteratorBrandCheckException('next');
+    }
+    try {
+      const reader = this._reader;
+      const read = GetMethod(reader, 'read');
+      return Reflect.apply(read, reader, []);
```

Comment:
Should we use [the `Call` helper function](https://github.com/whatwg/streams/blob/5d9498e7526d2f50618432e7aada8582d08745c0/reference-implementation/lib/helpers.js#L55) here? The reference implementation doesn't seem to use `Reflect` anywhere else, although I'm not sure whether that's a design goal.